packages/salesforcedx-vscode-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -75,16 +101,172 @@ export class CheckpointService implements TreeDataProvider<BaseNode> {
     return element.getChildren();
   }
 
-  public addCheckpointNode(
-    sourceFile: string,
-    typeRef: string,
-    line: number
-  ): CheckpointNode {
-    const cpNode = new CheckpointNode(sourceFile, typeRef, line);
+  // There's a limit to the number of checkpoints for a given user and
+  // that number is 5.
+  public canAddCheckpointNote(): Boolean {
+    return this.getChildren().length < MAX_ALLOWED_CHECKPOINTS;
+  }
+
+  public async addCheckpointNode(
+    uriInput: string,
+    sourceFileInput: string,
+    typeRefInput: string,
+    lineInput: number,
+    makeCommandCall = true
+  ): Promise<CheckpointNode> {
+    // Before creating a new checkpoint node, check and see if one with the
+    // URI and line number already exists. If one already exists then return
+    // that one, otherwise create one, add it to the list and then return it
+    const cpTemp = this.returnCheckpointNodeIfAlreadyExists(
+      uriInput,
+      lineInput
+    );
+    if (cpTemp) {
+      return cpTemp;
+    }
+    const cpNode = new CheckpointNode(
+      uriInput,
+      sourceFileInput,
+      typeRefInput,
+      lineInput
+    );
+    // This is here for testing purposes. The command is being tested separately
+    if (makeCommandCall === true) {
```

Comment:
This makeCommandCall might not be necessary. I read over your tests that use this and I think you could stub out this method to achieve the same result without a boolean parameter

Reply:
And also, for JavaScript and TypeScript, we don't check if something === true, we just do `if(makeCommandCall)`

Reply:
Fixed. Changed things to a stub and removed makeCommandCall altogether.